### ISSUE TITLE
Uneven recording size getting generated

### ISSUE THREAD

**Issue by divdaisymuffin** [OpenVisualCloud/Smart-City-Sample#793]
Hi @nnshah1 
We are facing an issue where when we specify the "max-size-time" field in pipeline.json in nanoseconds, we are observing that the size of recording getting generated are not same, sometimes they are less duration and sometimes they are of high duration.

Please let us know how to get uniform size recordings.

**Comment by divdaisymuffin** [comment]
@nnshah1 @xwu2git we got another issue while running pipeline on a live rtsp camera, Please look at the error below

`Error on Pipeline 1: gst-stream-error-quark: Could not multiplex stream. (10): ../gst/isomp4/gstqtmux.c(4832): gst_qt_mux_add_buffer (): /GstPipeline:pipeline4/GstSplitMuxSink:splitmuxsink/GstMP4Mux:muxer:\nBuffer has no PTS."`

which says specifically "buffer has no PTS", this we have faced first time only on this camera, it is running fine on another camera.

What is a alternate way to create a recording without PTS, Can we do it using gstreamer some parameter tuning?

Pipeline template we are using
`"template":"rtspsrc udp-buffer-size=212992 name=source ! queue ! rtph264depay ! h264parse ! video/x-h264 ! tee name=t ! queue ! decodebin ! videoconvert name=\"videoconvert\" ! video/x-raw,format=BGRx ! queue leaky=upstream ! gvadetect ie-config=CPU_BIND_THREAD=NO model=\"{models[head_yolov4_tiny_608to416_default_anchors_mask_012_INT8][1][network]}\" model-proc=\"{models[head_yolov4_tiny_608to416_default_anchors_mask_012_INT8][1][proc]}\" name=\"detection\" threshold=0.30 ! gvametaconvert name=\"metaconvert\" ! queue ! gvapython name=\"new_wait\" module=\"custom_transforms/new_wait\" class=\"WaitTime\" ! gvametapublish name=\"destination\" ! appsink name=appsink t. ! queue ! splitmuxsink max-size-time=300000000000 name=\"splitmuxsink\""`

**Comment by nnshah1** [comment]
@divdaisymuffin Can you double check the properties of the camera in terms of encoding - that it is h264, etc. - also does the pipeline work if you remove the splitmuxsink branch?

**Comment by divdaisymuffin** [comment]
@nnshah1 yes we checked camera property, it is h264 only, also confirmed by playing same rtspurl on vlc player and checked codec information, it is h264 only.
If we remove splitmuxsink recordings will not be able to create right?

**Comment by nnshah1** [comment]
@divdaisymuffin That's correct - I wanted to just confirm if the pipeline operates normally for detection and the problem is isolated to recording. That could give us an idea of what to look at.

Can you also share the make / model of the camera just in case we can find something related to that?

**Comment by divdaisymuffin** [comment]
NVR Details
 Device Type/model : DHI-NVR4432-4ks2

Camera Details
 
 Device Type
 DH-IPC-HDW1230T1P-S4
 System Version
 V2.680.0000000.7.R, Build Date: 2019-09-11
 WEB Version
 V3.2.1.26770
 ONVIF Version
 18.12(V2.0.0.26587)
 S/N
 5M0714BPAGE38BA
 Security Baseline Version
 V1.4
 Copyright 2019, all rights reserved.

**Comment by divdaisymuffin** [comment]
@nnshah1 Analytics is working fine, issue is with recording where in recording filename the offset value is _0 continuously. For eg. File name is 1634568904331_0.mp4
And it tries to create multiple files but all with _0.mp4

**Comment by nnshah1** [comment]
Are the timestamps in the metadata also all at 0? 

Is it possible to reproduce this from a recorded file (that is record via vlc and then run through pipeline directly with a filesrc?)

**Comment by divdaisymuffin** [comment]
@nnshah1 We found issue with the camera
The RTP packets miss proper information like DTS, Please see the attached snap.
![Screenshot (127)](https://user-images.githubusercontent.com/68895288/144874009-0e0f2aac-69f7-444d-96f9-419155b5b6e4.png)

Also it seems like, the splitmuxsink is very dependent on these informations from RTP packets? Do you know any other plugin which can create recording by neglecting DTS and PTS information

**Comment by nnshah1** [comment]
one thing you could try is to add a videorate element before the splitmuxsink to add in pts - I'd still like ot understand what timestamps the metadata are reporting - as you'll want to ensure you can synchronize those. in your gvapython code - you can get access to the buffer and print the .pts and .dts yourself - you could also potentially modify it

**Comment by nnshah1** [comment]
@divdaisymuffin We're not able to easily reproduce the issue you're seeing or offer a solution. Is there a way to reproduce without access to the camera in question? Also I believe we would want to open a seperate issue for this specific case (missing PTS / DTS). Let's use this issue to focus on normalizing recording segment size.
